lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -248,16 +256,16 @@ hasValidCodingKeysEnum(TypeChecker &tc, NominalTypeDecl *type,
 ///
 /// \param tc The typechecker to use in validating {En,De}codable conformance.
 ///
-/// \param type The nominal type decl whose nested \c CodingKeys type to
+/// \param target The nominal type decl whose nested \c CodingKeys type to
 /// synthesize.
 ///
 /// \param proto The {En,De}codable protocol to validate all the keys conform
 /// to.
 static EnumDecl *synthesizeCodingKeysEnum(TypeChecker &tc,
-                                          NominalTypeDecl *type,
+                                          NominalTypeDecl *target,
                                           ProtocolDecl *proto) {
   auto &C = tc.Context;
-  auto *typeDC = cast<DeclContext>(type);
+  auto *targetDC = cast<DeclContext>(target);
```

Comment:
This will definitely need to change in the future—if you add the conformance in an extension, the synthesized declarations go in that extension, not the original type.